File: Basic Nico variation/basic-nico-variation.py
def nico(key, message):

    length_key, len_msg = len(key), len(message)
    # get array new key position
    keys = [''.join(sorted(key)).find(x) for x in key]
    # add need some spaces to end of message
    message += (" " * (length_key - (len_msg % length_key) if len_msg % length_key != 0 else 0))
    # slice message by length key
    slice_messages = [message[start:start + length_key] for start in range(0, len(message), length_key)]

    new_message = ""
    for values in slice_messages:
        # sort dict by key and concat to new message
        new_message += "".join((x[1]) for x in sorted(dict(zip(keys, values)).items()))

    return new_message


print(nico("crazy", "secretinformation"))  # "cseerntiofarmit on  "

File: Basic Nico variation/basic-nico-variation.php
<?php
/**
 * Created by PhpStorm.
 * User: kovalevcon
 * Date: 25.09.18
 * Time: 17:27
 */

/**
 * Basic Nico variation function
 *
 * @param string $key
 * @param string $message
 * @return string
 */
function nico(string $key, string $message): string
{
    $sortKey = str_split($key);
    sort($sortKey);

    $result = "";
    for($i = 0, $lengthKey = strlen($key), $lengthMsg = strlen($message); $i < $lengthMsg; $i += $lengthKey) {
        foreach ($sortKey as $x) {
            $temp = mb_substr($message, $i, $i + $lengthKey);
            $result .= $temp[strpos($key, $x)] ?? ' ';
        }
    }

    return $result;
}

echo "<pre>nico('crazy', 'secretinformation');\r\n";
echo nico('crazy', 'secretinformation'); // 'cseerntiofarmit on  '
echo "</pre>";
